backend/btrixcloud/operator/crawls.py

```diff
@@ -410,8 +410,8 @@ async def _resolve_scale(
             actual_scale -= 1
 
         # ensure at least enough pages for the scale
-        if status.pagesFound and status.pagesFound < desired_scale:
-            desired_scale = status.pagesFound
+        if status.pagesFound < desired_scale:
+            desired_scale = max(1, status.pagesFound)
```

Comment:
Might this not set `scale == status.pagesFound`, even if that's higher than the scale that was set in the workflow?

Reply:
Though it seems that's how it worked before, maybe I'm just not grasping the purpose of `desired_scale`?

Reply:
Yes, the desired_scale is the scale from the workflow, so this only applies if pages found < scale, so if scale == 3, and pages found is 0, it will be set to 1, if its 2, it will be set to 2. Once pagesFound exceeds 3, it will remain at 3.

Reply:
The specific fix here was to account for status.pagesFound being 0, which ignored this condition, and resulted in setting a scale of 3 immediately before there were any pages, and then back to 1 if pagesFound was 1.

Reply:
> Once pagesFound exceeds 3, it will remain at 3.

Will it? I'm not seeing the logic that would cap `desired_scale` at the workflow scale here since we're overwriting `desired_scale` with the number of pages found, though I'm likely missing something

Reply:
Oh it's the if condition right above haha. Nevermind, ignore me, this looks great!

Reply:
The desired_scale is passed in from calling function from `CrawlJob.spec.scale`, which is immutable on the CrawlJob, so its just:
```
if pagesFound (0) < crawl scale (3)
   crawl scale (3) = max(1, pageFound (0))
```